engine/src/main/java/org/camunda/bpm/engine/impl/persistence/entity/util/TypedValueField.java

```diff
@@ -152,8 +152,12 @@ protected boolean isMutableValue(TypedValue value) {
     return((TypedValueSerializer<TypedValue>) getSerializer()).isMutableValue(value);
   }
 
+  private static boolean isImplicitVariableUpdateDetectionEnabled() {
+    return Context.getProcessEngineConfiguration().isImplicitVariableUpdateDetectionEnabled();
+  }
+
   protected boolean isValuedImplicitlyUpdated() {
-    if (cachedValue != null && isMutableValue(cachedValue)) {
+    if (cachedValue != null && isImplicitVariableUpdateDetectionEnabled() && isMutableValue(cachedValue)) {
```

Comment:
I suggest to move this condition to `#notifyImplicitValueUpdate`. This seems a little more fitting to me, as `#isValuedImplicitlyUpdated` sounds like it returns the value independent of configuration. Maybe rename `notifyImplicitValueUpdate` to `notifyImplicitValueUpdateIfEnabled` if we want to be most precise.

Reply:
Sounds good. Will do.